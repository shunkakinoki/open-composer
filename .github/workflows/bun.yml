name: Bun
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  workflow_dispatch:
concurrency:
  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.head_ref || github.sha }}
  cancel-in-progress: true
jobs:
  bun-build:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Build
        run: bun run build
  bun-lint:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Check
        run: bun run check
      - name: Format
        run: bun run format
      - name: Lint
        run: bun run lint
      - name: Type Check
        run: bun run type-check
  bun-test:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Test
        run: bun run test
  bun-check:
    if: always()
    needs:
      - bun-build
      - bun-lint
      - bun-test
    runs-on: ubuntu-latest
    timeout-minutes: 3
    steps:
      - name: Alls Green
        uses: re-actors/alls-green@release/v1
        with:
          jobs: ${{ toJSON(needs) }}
