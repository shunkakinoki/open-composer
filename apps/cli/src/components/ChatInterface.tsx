import { TextAttributes } from "@opentui/core";
import { useKeyboard } from "@opentui/react";
import type React from "react";
import { useState } from "react";

interface ChatMessage {
  id: string;
  content: string;
  sender: "user" | "agent";
  agent?: string;
  timestamp: Date;
}

interface ChatInterfaceProps {
  messages: ChatMessage[];
  onSendMessage: (message: string) => void;
}

export const ChatInterface: React.FC<ChatInterfaceProps> = ({
  messages,
  onSendMessage,
}) => {
  const [input, setInput] = useState("");

  useKeyboard((key) => {
    if (key.name === "return") {
      if (input.trim()) {
        onSendMessage(input.trim());
        setInput("");
      }
    } else if (key.name === "backspace" || key.name === "delete") {
      setInput((prev) => prev.slice(0, -1));
    } else if (key.sequence && key.sequence.length === 1 && !key.ctrl && !key.meta) {
      setInput((prev) => prev + key.sequence);
    }
  });

  const formatTime = (date: Date) => {
    return date.toLocaleTimeString("en-US", {
      hour12: false,
      hour: "2-digit",
      minute: "2-digit",
    });
  };

  const getAgentIcon = (agent?: string) => {
    switch (agent) {
      case "claude-code":
        return "🤖";
      case "codex-nation":
        return "📝";
      case "cursor-agent":
        return "🖱️";
      default:
        return "🎼";
    }
  };

  return (
    <box style={{ flexDirection: "column", padding: 1 }}>
      <text
        content="💬 Chat Interface"
        style={{ fg: "green", attributes: TextAttributes.BOLD }}
      />

      <box style={{ flexDirection: "column", flexGrow: 1, marginTop: 1 }}>
        {messages.map((message) => (
          <box key={message.id} style={{ flexDirection: "column", marginBottom: 1 }}>
            <box>
              <text
                content={`${message.sender === "user" ? "👤" : getAgentIcon(message.agent)} ${message.sender === "user" ? "You" : message.agent || "Agent"}`}
                style={{ fg: message.sender === "user" ? "cyan" : "yellow" }}
              />
              <text
                content={` ${formatTime(message.timestamp)}`}
                style={{ fg: "gray" }}
              />
            </box>
            <text content={` ${message.content}`} />
          </box>
        ))}
      </box>

      <box style={{ border: true, borderColor: "blue", marginTop: 1 }}>
        <text content="💭 " style={{ fg: "blue" }} />
        <text content={input} />
        <text content="_" style={{ fg: "gray" }} />
      </box>

      <text content="Type your message and press Enter..." style={{ fg: "gray" }} />
    </box>
  );
};
