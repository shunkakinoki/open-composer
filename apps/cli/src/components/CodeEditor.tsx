// biome-ignore assist/source/organizeImports: False positive
import type React from "react";
import { Box, Text } from "ink";

interface CodeEditorProps {
  currentFile?: string;
}

export const CodeEditor: React.FC<CodeEditorProps> = ({ currentFile }) => {
  return (
    <Box flexDirection="column" padding={1}>
      <Text bold color="magenta">
        📝 Code Editor
      </Text>

      <Box flexDirection="column" marginTop={1}>
        {currentFile ? (
          <>
            <Text color="blue">📄 {currentFile}</Text>
            <Box
              borderStyle="single"
              borderColor="gray"
              padding={1}
              marginTop={1}
              flexGrow={1}
            >
              <Text color="gray" dimColor>
                Code content will appear here...
              </Text>
            </Box>
          </>
        ) : (
          <Box
            borderStyle="single"
            borderColor="gray"
            padding={2}
            flexGrow={1}
            justifyContent="center"
            alignItems="center"
          >
            <Text color="gray" dimColor>
              No file selected
            </Text>
          </Box>
        )}
      </Box>

      <Box marginTop={1}>
        <Text bold color="yellow">
          🔧 File Tree
        </Text>
      </Box>

      <Box flexDirection="column" marginTop={1}>
        <Text color="gray">📁 src/</Text>
        <Text color="gray">{"  "}📄 index.ts</Text>
        <Text color="gray">{"  "}📁 components/</Text>
        <Text color="gray">{"    "}📄 ComposerApp.tsx</Text>
        <Text color="gray">{"    "}📄 ChatInterface.tsx</Text>
      </Box>
    </Box>
  );
};
