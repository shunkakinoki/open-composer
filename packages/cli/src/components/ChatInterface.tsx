// biome-ignore assist/source/organizeImports: False positive
import type React from "react";
import { useState } from "react";
import { Box, Text, useInput } from "ink";

interface ChatMessage {
  id: string;
  content: string;
  sender: "user" | "agent";
  agent?: string;
  timestamp: Date;
}

interface ChatInterfaceProps {
  messages: ChatMessage[];
  onSendMessage: (message: string) => void;
}

export const ChatInterface: React.FC<ChatInterfaceProps> = ({
  messages,
  onSendMessage,
}) => {
  const [input, setInput] = useState("");
  const [_isTyping, _setIsTypingg] = useState(false);

  useInput((input, key) => {
    if (key.return) {
      if (input.trim()) {
        onSendMessage(input.trim());
        setInput("");
      }
    } else if (key.backspace || key.delete) {
      setInput((prev) => prev.slice(0, -1));
    } else if (input.length === 1 && !key.ctrl && !key.meta) {
      setInput((prev) => prev + input);
    }
  });

  const formatTime = (date: Date) => {
    return date.toLocaleTimeString("en-US", {
      hour12: false,
      hour: "2-digit",
      minute: "2-digit",
    });
  };

  const getAgentIcon = (agent?: string) => {
    switch (agent) {
      case "claude-code":
        return "🤖";
      case "codex-nation":
        return "📝";
      case "cursor-agent":
        return "🖱️";
      default:
        return "🎼";
    }
  };

  return (
    <Box flexDirection="column" padding={1}>
      <Text bold color="green">
        💬 Chat Interface
      </Text>

      <Box flexDirection="column" flexGrow={1} marginTop={1}>
        {messages.map((message) => (
          <Box key={message.id} flexDirection="column" marginBottom={1}>
            <Box>
              <Text color={message.sender === "user" ? "cyan" : "yellow"}>
                {message.sender === "user" ? "👤" : getAgentIcon(message.agent)}{" "}
                {message.sender === "user" ? "You" : message.agent || "Agent"}
              </Text>
              <Text color="gray" dimColor>
                {" "}
                {formatTime(message.timestamp)}
              </Text>
            </Box>
            <Text> {message.content}</Text>
          </Box>
        ))}
      </Box>

      <Box borderStyle="single" borderColor="blue" marginTop={1}>
        <Text color="blue">💭 </Text>
        <Text>{input}</Text>
        <Text color="gray" dimColor>
          _
        </Text>
      </Box>

      <Text color="gray" dimColor>
        Type your message and press Enter...
      </Text>
    </Box>
  );
};
