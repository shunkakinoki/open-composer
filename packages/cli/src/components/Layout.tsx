// biome-ignore assist/source/organizeImports: False positive
import type React from "react";
import { Box, Text } from "ink";

interface LayoutProps {
  children: React.ReactNode;
}

export const Layout: React.FC<LayoutProps> = ({ children }) => {
  return (
    <Box flexDirection="column" height="100%">
      <Box borderStyle="single" borderColor="blue" justifyContent="center">
        <Text color="blue" bold>
          ðŸŽ¼ Open Composer CLI v0.1.0
        </Text>
      </Box>

      <Box flexGrow={1}>{children}</Box>

      <Box
        borderStyle="single"
        borderColor="gray"
        justifyContent="space-between"
        flexDirection="row"
      >
        <Text color="gray">Ready</Text>
        <Text color="gray">Ctrl+C to exit</Text>
      </Box>
    </Box>
  );
};
