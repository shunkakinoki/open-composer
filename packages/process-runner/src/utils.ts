import * as Effect from "effect/Effect";
import {
  MAX_COMMAND_LENGTH,
  MAX_RUN_NAME_LENGTH,
  RUN_NAME_PATTERN,
} from "./constants.js";
import { ProcessRunnerError } from "./types.js";

// =============================================================================
// VALIDATION UTILITIES
// =============================================================================

export function validateRunName(
  runName: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!runName || typeof runName !== "string") {
    return Effect.fail(
      ProcessRunnerError("Run name must be a non-empty string"),
    );
  }

  if (runName.length > MAX_RUN_NAME_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Run name too long (max ${MAX_RUN_NAME_LENGTH} characters)`,
      ),
    );
  }

  if (!RUN_NAME_PATTERN.test(runName)) {
    return Effect.fail(
      ProcessRunnerError(
        "Run name can only contain letters, numbers, hyphens, and underscores",
      ),
    );
  }

  return Effect.succeed(runName);
}

export function validateCommand(
  command: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!command || typeof command !== "string") {
    return Effect.fail(
      ProcessRunnerError("Command must be a non-empty string"),
    );
  }

  if (command.length > MAX_COMMAND_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Command too long (max ${MAX_COMMAND_LENGTH} characters)`,
      ),
    );
  }

  // Basic command injection prevention - only reject null bytes
  if (command.includes("\x00")) {
    return Effect.fail(
      ProcessRunnerError("Command contains invalid characters"),
    );
  }

  return Effect.succeed(command);
}

// =============================================================================
// UTILITY FUNCTIONS
// =============================================================================

export function withTimeout<T>(
  effect: Effect.Effect<T, ProcessRunnerError>,
  timeoutMs: number,
  timeoutMessage: string = "Operation timed out",
): Effect.Effect<T, ProcessRunnerError> {
  const timeoutEffect = Effect.async<T, ProcessRunnerError>((resume) => {
    const timeout = setTimeout(() => {
      resume(Effect.fail(ProcessRunnerError(timeoutMessage)));
    }, timeoutMs);
    return Effect.sync(() => clearTimeout(timeout));
  });

  return Effect.race(effect, timeoutEffect);
}
