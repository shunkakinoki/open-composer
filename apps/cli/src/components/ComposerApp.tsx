import { Box } from "ink";
import type React from "react";
import { useState } from "react";
import { ChatInterface } from "./ChatInterface.js";
import { CodeEditor } from "./CodeEditor.js";
import { Layout } from "./Layout.js";
import { Sidebar } from "./Sidebar.js";

// Utility function for creating timestamps - can be mocked in tests
export const createTimestamp = () => new Date();
export const getCurrentTime = () => Date.now();

export interface ComposerAppState {
  currentBranch: string;
  currentFile?: string;
  chatMessages: Array<{
    id: string;
    content: string;
    sender: "user" | "agent";
    agent?: string;
    timestamp: Date;
  }>;
  worktrees: Array<{
    name: string;
    path: string;
    branch: string;
    active: boolean;
  }>;
}

export const ComposerApp: React.FC = () => {
  const [state, setState] = useState<ComposerAppState>({
    currentBranch: "main",
    chatMessages: [
      {
        id: "1",
        content: "Welcome to Open Composer CLI! What would you like to build?",
        sender: "agent",
        agent: "composer",
        timestamp: createTimestamp(),
      },
    ],
    worktrees: [
      {
        name: "main",
        path: process.cwd(),
        branch: "main",
        active: true,
      },
    ],
  });

  const handleSendMessage = (message: string) => {
    const newMessage = {
      id: getCurrentTime().toString(),
      content: message,
      sender: "user" as const,
      timestamp: createTimestamp(),
    };

    setState((prev) => ({
      ...prev,
      chatMessages: [...prev.chatMessages, newMessage],
    }));

    // Mock agent response
    setTimeout(() => {
      const agentResponse = {
        id: (getCurrentTime() + 1).toString(),
        content: `I'll help you with: "${message}". Let me analyze your request...`,
        sender: "agent" as const,
        agent: "claude-code",
        timestamp: createTimestamp(),
      };

      setState((prev) => ({
        ...prev,
        chatMessages: [...prev.chatMessages, agentResponse],
      }));
    }, 1000);
  };

  return (
    <Layout>
      <Box flexDirection="row" height="100%">
        <Box width="30%" borderStyle="single" borderColor="gray">
          <Sidebar
            worktrees={state.worktrees}
            currentBranch={state.currentBranch}
          />
        </Box>

        <Box width="40%" borderStyle="single" borderColor="gray">
          <ChatInterface
            messages={state.chatMessages}
            onSendMessage={handleSendMessage}
          />
        </Box>

        <Box width="30%" borderStyle="single" borderColor="gray">
          <CodeEditor currentFile={state.currentFile} />
        </Box>
      </Box>
    </Layout>
  );
};
