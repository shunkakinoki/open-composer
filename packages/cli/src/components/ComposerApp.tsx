// biome-ignore assist/source/organizeImports: False positive
import type React from "react";
import { useState } from "react";
import { Box } from "ink";
import { Layout } from "./Layout.js";
import { ChatInterface } from "./ChatInterface.js";
import { Sidebar } from "./Sidebar.js";
import { CodeEditor } from "./CodeEditor.js";

export interface ComposerAppState {
  currentBranch: string;
  currentFile?: string;
  chatMessages: Array<{
    id: string;
    content: string;
    sender: "user" | "agent";
    agent?: string;
    timestamp: Date;
  }>;
  worktrees: Array<{
    name: string;
    path: string;
    branch: string;
    active: boolean;
  }>;
}

export const ComposerApp: React.FC = () => {
  const [state, setState] = useState<ComposerAppState>({
    currentBranch: "main",
    chatMessages: [
      {
        id: "1",
        content: "Welcome to Open Composer CLI! What would you like to build?",
        sender: "agent",
        agent: "composer",
        timestamp: new Date(),
      },
    ],
    worktrees: [
      {
        name: "main",
        path: process.cwd(),
        branch: "main",
        active: true,
      },
    ],
  });

  const handleSendMessage = (message: string) => {
    const newMessage = {
      id: Date.now().toString(),
      content: message,
      sender: "user" as const,
      timestamp: new Date(),
    };

    setState((prev) => ({
      ...prev,
      chatMessages: [...prev.chatMessages, newMessage],
    }));

    // Mock agent response
    setTimeout(() => {
      const agentResponse = {
        id: (Date.now() + 1).toString(),
        content: `I'll help you with: "${message}". Let me analyze your request...`,
        sender: "agent" as const,
        agent: "claude-code",
        timestamp: new Date(),
      };

      setState((prev) => ({
        ...prev,
        chatMessages: [...prev.chatMessages, agentResponse],
      }));
    }, 1000);
  };

  return (
    <Layout>
      <Box flexDirection="row" height="100%">
        <Box width="30%" borderStyle="single" borderColor="gray">
          <Sidebar
            worktrees={state.worktrees}
            currentBranch={state.currentBranch}
          />
        </Box>

        <Box width="40%" borderStyle="single" borderColor="gray">
          <ChatInterface
            messages={state.chatMessages}
            onSendMessage={handleSendMessage}
          />
        </Box>

        <Box width="30%" borderStyle="single" borderColor="gray">
          <CodeEditor currentFile={state.currentFile} />
        </Box>
      </Box>
    </Layout>
  );
};
