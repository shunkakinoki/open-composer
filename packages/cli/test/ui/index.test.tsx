import { describe, expect, mock, test } from "bun:test";
import { Text } from "ink";

// Mock ComposerApp utility functions before importing
const mockCreateTimestamp = mock(() => new Date("2024-01-01T10:00:00Z"));
const mockGetCurrentTime = mock(() =>
  new Date("2024-01-01T10:00:00Z").getTime(),
);

mock.module("../../src/components/ComposerApp.js", () => ({
  createTimestamp: mockCreateTimestamp,
  getCurrentTime: mockGetCurrentTime,
}));

import { ChatInterface } from "../../src/components/ChatInterface.js";
import { CodeEditor } from "../../src/components/CodeEditor.js";
import { ComposerApp } from "../../src/components/ComposerApp.js";
import { Layout } from "../../src/components/Layout.js";
import { Sidebar } from "../../src/components/Sidebar.js";
import { AgentRouter, WorktreeManager } from "../../src/lib/index.js";
import { render } from "../utils.js";

describe("Open Composer CLI", () => {
  describe("AgentRouter", () => {
    test("should initialize with default agents", () => {
      const router = new AgentRouter();
      const agents = router.getAgents();

      expect(agents).toHaveLength(5);
      expect(agents.find((a) => a.name === "claude-code")).toBeDefined();
      expect(agents.find((a) => a.name === "codex-nation")).toBeDefined();
    });

    test("should activate and deactivate agents", () => {
      const router = new AgentRouter();

      expect(router.activateAgent("codex-nation")).toBe(true);
      expect(router.deactivateAgent("claude-code")).toBe(true);

      const activeAgents = router.getActiveAgents();
      expect(activeAgents.find((a) => a.name === "codex-nation")).toBeDefined();
      expect(
        activeAgents.find((a) => a.name === "claude-code"),
      ).toBeUndefined();
    });

    test("should route queries to appropriate agents", async () => {
      const router = new AgentRouter();

      const reviewResponse = await router.routeQuery("review this code");
      expect(reviewResponse.agent).toBe("claude-code");

      const generateResponse = await router.routeQuery("generate a function");
      expect(generateResponse.agent).toBe("codex-nation");
    });
  });

  describe("WorktreeManager", () => {
    test("should initialize with current directory", () => {
      const manager = new WorktreeManager();
      expect(manager).toBeDefined();
    });

    test("should get current branch", async () => {
      const manager = new WorktreeManager();
      const branch = await manager.getCurrentBranch();
      expect(typeof branch).toBe("string");
    });
  });

  describe("Component Snapshots", () => {
    test("ComposerApp renders correctly", () => {
      const { lastFrame } = render(<ComposerApp />);
      expect(lastFrame()).toMatchSnapshot();
    });

    test("ChatInterface renders correctly with messages", () => {
      const messages = [
        {
          id: "1",
          content: "Hello, how can I help you?",
          sender: "agent" as const,
          agent: "claude-code",
          timestamp: new Date("2024-01-01T10:00:00"),
        },
        {
          id: "2",
          content: "I need help with my code",
          sender: "user" as const,
          timestamp: new Date("2024-01-01T10:01:00"),
        },
      ];

      const { lastFrame } = render(
        <ChatInterface messages={messages} onSendMessage={() => {}} />,
      );
      expect(lastFrame()).toMatchSnapshot();
    });

    test("Sidebar renders correctly with worktrees", () => {
      const worktrees = [
        {
          name: "main",
          path: "/path/to/main",
          branch: "main",
          active: true,
        },
        {
          name: "feature-1",
          path: "/path/to/feature-1",
          branch: "feature-1",
          active: false,
        },
      ];

      const { lastFrame } = render(
        <Sidebar worktrees={worktrees} currentBranch="main" />,
      );
      expect(lastFrame()).toMatchSnapshot();
    });

    test("CodeEditor renders correctly with file", () => {
      const { lastFrame } = render(<CodeEditor currentFile="src/index.ts" />);
      expect(lastFrame()).toMatchSnapshot();
    });

    test("CodeEditor renders correctly without file", () => {
      const { lastFrame } = render(<CodeEditor />);
      expect(lastFrame()).toMatchSnapshot();
    });

    test("Layout renders correctly", () => {
      const { lastFrame } = render(
        <Layout>
          <Text>Test content</Text>
        </Layout>,
      );
      expect(lastFrame()).toMatchSnapshot();
    });
  });
});
