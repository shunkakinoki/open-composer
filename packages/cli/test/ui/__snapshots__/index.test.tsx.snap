// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`Open Composer CLI Component Snapshots ComposerApp renders correctly 1`] = `
"┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                   🎼 Open Composer CLI v0.1.0                                    │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
┌────────────────────────────┐┌──────────────────────────────────────┐┌────────────────────────────┐
│                            ││                                      ││                            │
│ 📂 Workspaces              ││ 💬 Chat Interface                    ││ 📝 Code Editor             │
│                            ││                                      ││                            │
│ ● main                     ││ 🎼 composer 14:55                    ││ ┌──────────────────────┐   │
│                            ││  Welcome to Open Composer CLI! What  ││ │                      │   │
│                            ││ would you like to build?             ││ │                      │   │
│ 🌿 Branches                ││                                      ││ │   No file selected   │   │
│                            ││                                      ││ │                      │   │
│ ● main                     ││                                      ││ │                      │   │
│ + New worktree             ││                                      ││ └──────────────────────┘   │
│                            ││                                      ││                            │
│                            ││                                      ││ 🔧 File Tree               │
│ 🤖 Agents                  ││                                      ││                            │
│                            ││                                      ││ 📁 src/                    │
│ ● claude-code              ││ ┌──────────────────────────────────┐ ││   📄 index.ts              │
│ ○ codex-nation             ││ │💭 _                              │ ││   📁 components/           │
│ ○ cursor-agent             ││ └──────────────────────────────────┘ ││     📄 ComposerApp.tsx     │
│                            ││ Type your message and press Enter... ││     📄 ChatInterface.tsx   │
│                            ││                                      ││                            │
└────────────────────────────┘└──────────────────────────────────────┘└────────────────────────────┘
┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│Ready                                                                               Ctrl+C to exit│
└──────────────────────────────────────────────────────────────────────────────────────────────────┘"
`;
